lib/puppet/file_serving/http_metadata.rb

```diff
@@ -0,0 +1,29 @@
+require 'puppet/file_serving/metadata'
+
+# Simplified metadata representation, suitable for the information
+# that is available from HTTP headers.
+class Puppet::FileServing::HttpMetadata < Puppet::FileServing::Metadata
+
+  def initialize(http_response, path = '/dev/null')
+    super(path)
+
+    # ignore options that do not apply to HTTP metadata
+    @owner = @group = @mode = nil
+
+    if checksum = http_response['content-md5']
+      # convert base64 digest to hex
+      checksum = checksum.unpack("m0").first.unpack("H*").first
+      @checksum_type = 'md5'
+      @checksum = "{md5}#{checksum}"
+    elsif mtime = DateTime.httpdate(http_response['last-modified'])
+      @checksum_type = 'mtime'
+      @checksum = "{mtime}#{mtime}"
```

Comment:
Specifically, this needs to produce a representation that is compatible.

The local `mtime` is retrieved by `FileServing::Metadata` through `Util::Checksums::mtime_file` which in turn relies on `File::Stat::mtime` (surprise, surprise).

Quoth the reference, this method returns "`aTIme`", which I assume will be a `Time` object rather than not. So perhaps we should just convert the `DateTIme` to a `Time` here. I have used the library class `DateTime` because it can parse HTTP dates, which core `Time` cannot, apparently.